forem/forem
# render left sidebar billboards ('display ads') asynchronously

## What type of PR is this? (check all applicable)


## Description

Given all the permutations of how a given billboard (with or without a given org affiliation) might be targeted to a given user, and how those possibilities affect the complexity of managing cache concerns, we would like to be able to serve more of our billboards asynchronously.

This change updates how the `sidebar_left` and `sidebar_left_2` ads are rendered. It also removes the tests for the presence of those ads at the time of the server's response to the stories index request, since the ads haven't showed up at that time.

## Related Tickets & Documents



- Related Issue #19577
- Closes #19577

## QA Instructions, Screenshots, Recordings

If display ads do not show up in the sidebar in your Uffizzi instance:
- log in as admin
- visit /admin and navigate to Customization -> Display Ads
- create ads that are both Published: true and Approved: true for placements Sidebar Left and Sidebar Left 2
- visit home feed again to view the ads, and check for padding above and between them
- *known issue* is that the dropdown in a very short display ad with no image is larger than the ad that contains it. Have not addressed that yet, may deserve its own style bug issue

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

      have not been included_
- [X] I need help with writing tests (eager for team discussion on best e2e approach for billboard testing)

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![Tabletop RPG gamer saying "you have to respect the sidebar"](https://media1.giphy.com/media/cKQE7r0Tk1QEwUg1n6/giphy.gif)
